### PR TITLE
toggle-button-group: reworked responsiveness and added equal button heights

### LINKED ISSUE #2310: toggle-button-group: add equal heights and make fully responsive
### Description

Currently, our `toggle-buttons` within a `toggle-button-group` do not have the same height so when the content of one button is taller than others, the layout is not balanced.

Additionally, the initial iteration of `toggle-button-group` was semi-responsive with the use of 'desired columns` for the preferred number of buttons per row before wrapping, which was not guaranteed.

Since these two features are extremely intertwined because of a shift to use `display: grid` and both require a breaking change, it's best to do them at the same time.

For this issue, 

1. Add equal heights feature to `toggle-button-group` so that an increase in height of one button makes all buttons the same height, whether they are on a single row or span multiple rows.
2. Make `toggle-button-group` fully responsive using container queries.

### Screenshots

**Current:**

image

**Desired:**

image

### Figma link

_No response_

### Engineering pre-requisites

- [X] Successful design review with Design System Team.
- [X] Supports dark mode.
- [X] Supports responsive design.
- [X] Tokens have been provided (or already exist).
- [X] Successful accessibility review with Core A11Y Team.
- [X] Successful frontend review with eBayUI Team.

### PR DESCRIPTION
**Pull request by ArtBlue**
Fixes #2310 

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
This change,
1. Makes `toggle-button-group` fully responsive without user specified attributes, which avoids doing any calculations making things much simpler.
2. Makes all the `toggle-button` components inside `toggle-button-group` the same height - all buttons will adopt the height of the tallest button.
3. Per DS, the `list` layout toggle buttons need to now be top-aligned to make the buttons look good when taller.
4. Approved by DS, we are now being explicit about how many buttons should appear on a single line (before wrapping) at each breakpoint. I added those specifications to `css` props to make maintenance easier.
5. To allow users to implement some exceptions based on their own unique use cases, I added attribute hooks for specifying a specific number of buttons on each line before wrapping - `data-columns-xs` for the `XS` breakpoint, `data-columns-sm` for `SM`...
6. Modified docs to match the changes.
7. Added a responsive layout tool to docs to allow quick visualization of the responsive adjustments.

## Breaking Changes
1. The previously implemented user-specified responsive attribute, `data-columns`, was removed.
2. The top-level component element was moved from `ul.toggle-button-group` to `div.toggle-button-group` with the `ul` as the child. This was needed in order to use container queries, which requires a container of the element that needs responsive modifications.

## Notes
1. To ensure best responsive results, `max-width`, `min-width`, and `width` had to be changed from absolute pixels to relative units or removed altogether.
2. To allow `@container` queries, an override to `gulp-clean-css` was needed. 
3. To make the declarations of the classes for the exception cases much easier, I wrote a recursive mixin that also keeps the `less` file smaller. If we'd rather have the the raw `css`, we can just remove the mixin in lieu of the raw `css`.

## Screenshots
<kbd>toggle-button-group-responsive</kbd>

<kbd>toggle-button-group-equal-heights</kbd>

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Review by agliga — APPROVED** [review]

**Comment by agliga** [comment]
Lets fix the conflicts and we can merge.

**Comment by ArtBlue** [comment]
> Lets fix the conflicts and we can merge.

@agliga , can you take another pass and verify the sass conversion? I completely revamped the mixin and some of the vars needed different handling.

File: package.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
         "prerelease:end": "changeset pre exit || :"
     },
     "devDependencies": {
+        "@aptuitiv/gulp-clean-css": "^4.4.3",
```
Comment (agliga) [inline package.json:72]: Also remove

File: src/sass/toggle-button-group/toggle-button-group.scss [hunk 2]
```diff
@@ -1,7 +1,62 @@
...
+    // $selector: '.toggle-button-group[${attr}] ul';
+
+    @if $i <= $repeatTo {
+        .toggle-button-group[data-columns-#{$breakpoint}="#{$i}"] ul {
+            grid-template-columns: repeat($i, 1fr);
+        }
+
+        @include toggle-button-grid-templates($breakpoint, $repeatTo, ($i + 1));
+    }
+}
```
Comment (agliga) [inline src/sass/toggle-button-group/toggle-button-group.scss:15]: Should we move this a common place under `mixins` folder?

File: src/sass/toggle-button-group/toggle-button-group.scss [hunk 3]
```diff
@@ -89,23 +143,199 @@
...
+                var(--row-list-toggle-buttons-md),
+                1fr
+            );
+        }
+
+        .toggle-button-group--gallery-layout ul {
+            grid-template-columns: repeat(
+                var(--row-gallery-toggle-buttons-md),
+                1fr
+            );
```
Comment (agliga) [inline src/sass/toggle-button-group/toggle-button-group.scss:223]: Im wondering, why do we have this together with the mixins?

**Review by agliga — COMMENTED** [review]
Overall looks good, just a bit of cleanup and questions

File: package.json (hunk 1 above)
Comment (ArtBlue) [reply package.json:72]: Good catches on these packages. They're left over from pre-sass. I'll remove them.

File: src/sass/toggle-button-group/toggle-button-group.scss (hunk 2 above)
Comment (ArtBlue) [reply src/sass/toggle-button-group/toggle-button-group.scss:15]: I thought about that, but it's so specific to the toggle button groups and I can't think of any other use case for this be a general utility.

File: src/sass/toggle-button-group/toggle-button-group.scss (hunk 3 above)
Comment (ArtBlue) [reply src/sass/toggle-button-group/toggle-button-group.scss:223]: These are the internal `toggle-button-group` rules for prescribing how many buttons should be on each line before wrapping, and they're established by us. On the other hand, the classes generated by the mixin allow for the uses of the exception cases where the dev can use custom rules for specific cases. It's `#5` in the PR description.

**Comment by LuLaValva** [comment]
Is this intentional? Looks like gallery layout is vertically centered even though the rest are top aligned.
image representing a toggle button group (gallery layout) with one item that overflows its container. The other containers grow, and their contents are vertically centered.

**Comment by ArtBlue** [comment]
> Is this intentional? Looks like gallery layout is vertically centered even though the rest are top aligned.

Not intentional. Good catch!

File: package.json [hunk 4]
```diff
@@ -176,5 +176,6 @@
         "postcss-mixins": "^10.0.1",
         "postcss-nested": "^6.0.1",
         "postcss-simple-vars": "^7.0.1"
-    }
+    },
+    "packageManager": "pnpm@9.4.0+sha512.f549b8a52c9d2b8536762f99c0722205efc5af913e77835dbccc3b0b0b2ca9e7dc8022b78062c17291c48e88749c70ce88eb5a74f1fa8c4bf5e18bb46c8bd83a"
```
Comment (agliga) [inline package.json:180]: Hmm whats this?

**Review by agliga — COMMENTED** [review]
LGTM overall
Just saw some junk in package.json. Wondering if its intentional or accident.

File: package.json (hunk 4 above)
Comment (ArtBlue) [reply package.json:180]: I've no idea. I removed it, built locally, and it came back.

**Review by agliga — APPROVED** [review]

**Pull request merged**